File: problemsets/Codeforces/C++/A855.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include<bits/stdc++.h>
using namespace std;
int main() {
	int n; cin >> n;
	string s;
	set<string> a;
	while (n--) cin >> s, cout << (a.count(s)?"YES":(a.insert(s),"NO")) << endl;
}

File: problemsets/Codeforces/Python/A855.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
n,s=int(input()),{''}
for _ in [0]*n:
  a = input()
  print(['NO','YES'][a in s])
  s.add(a)